src/codeql.test.ts

```diff
@@ -89,3 +89,32 @@ test("running a query in a pack", async (t: any) => {
     await rmRF(tmpDir);
   }
 });
+
+test("getting the commit SHA for a database", async (t: any) => {
+  // Note: db-metadata-only.zip is not a real database.
+  // It doesn't contain source code, only a codeql-database.yml file.
+  const testDbZip = path.resolve("testdata/db-metadata-only.zip");
+  const tmpDir = fs.mkdtempSync("tmp");
+  const cwd = process.cwd();
+  process.chdir(tmpDir);
+  try {
+    // Test our fake database. This "database" has a commit SHA in its metadata.
+    await exec("codeql", ["database", "unbundle", testDbZip, "--name=fakeDb"]);
+    const sha1 = getDatabaseSHA("fakeDb");
+    t.is(sha1, "ccf1e13626d97b009b4da78f719f028d9f7cdf80");
+
+    // Test the real database, which has no commit SHA (until CodeQL CLI 2.7.2 is released).
+    // TODO: update this test once CodeQL CLI 2.7.2 is released.
```

Comment:
This is an interesting case because the test will suddenly start failing on `main` without us having taken any action. On a larger project this could really confuse other people, but here where all three of us are aware I think it's completely fine, and actually useful for us to know when the feature will start working.